pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -202,59 +209,43 @@ func ensure(clientset flowcontrolclient.FlowcontrolV1beta3Interface, fsLister fl
 }
 
 func ensureSuggestedConfiguration(clientset flowcontrolclient.FlowcontrolV1beta3Interface, fsLister flowcontrollisters.FlowSchemaLister, plcLister flowcontrollisters.PriorityLevelConfigurationLister) error {
-	plEnsurer := ensurer.NewSuggestedPriorityLevelEnsurerEnsurer(clientset.PriorityLevelConfigurations(), plcLister)
-	if err := plEnsurer.Ensure(flowcontrolbootstrap.SuggestedPriorityLevelConfigurations); err != nil {
+	plcSuggesteds := ensurer.WrapBootstrapPriorityLevelConfigurations(clientset.PriorityLevelConfigurations(), plcLister, flowcontrolbootstrap.SuggestedPriorityLevelConfigurations)
```

Comment:
is it possible to merge the suggested PLs and flowshemas into one slice, so we have one `EnsureConfigurations` call for all the sugested?

Reply:
That would be no small change.  Remember that `ensurer.WrapBootstrapPriorityLevelConfigurations` is not returning a slice but rather an interface with the following signature.

```
// BootstrapObjects is a generic interface to a list of bootstrap objects bound up with the relevant operations on them.
// The binding makes it unnecessary to have any type casts.
// A bootstrap object is a mandatory or suggested config object,
// with the spec that the code is built to provide.
type BootstrapObjects interface {
	typeName() string                         // the Kind of the objects
	len() int                                 // number of objects
	get(int) bootstrapObject                  // extract one object, origin 0
	getExistingObjects() ([]deletable, error) // returns all the APF config objects that exist at the moment
}
```

and `RemoveUnwantedObjects` assumes that they all have distinct names.